src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,166 @@
+import classNames from 'classnames';
+import { useState, useContext } from 'react';
+import { PatchedTodo, Todo } from '../../types/Todo';
+import { Errors } from '../../types/Errors';
+import { updateTodoTitle } from '../../api/todos';
+import { LoadingContext } from '../LoadingContext/LoadingContext';
+
+  type Props = {
+    todo: Todo
+    onChangeIsError: (e: Errors) => void
+    onDeleteTodo: (todoId: number) => void
+    onChangeTodo:
+    (id: number, data: PatchedTodo) => void;
+  };
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onChangeIsError,
+  onDeleteTodo,
+  onChangeTodo,
+}) => {
+  let { title } = todo;
+  const { id } = todo;
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+  const [isLoading, setIsLoading] = useState(id === 0);
+  const { isLoadingAll } = useContext(LoadingContext);
+
+  const handleDeleteTodo = async () => {
+    try {
+      setIsLoading(true);
+      await onDeleteTodo(todo.id);
+    } catch {
+      onChangeIsError(Errors.DELETE);
+      setIsLoading(false);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleChangeTitle = () => {
+    setIsEditing(true);
+  };
+
+  const cancelEditing = () => {
+    setIsEditing(false);
+    setNewTitle(title);
+  };
+
+  const updateTitle = async () => {
+    if (newTitle === title) {
+      await cancelEditing();
+
+      return;
+    }
+
+    if (!newTitle.trim()) {
+      await onDeleteTodo(todo.id);
+
+      return;
+    }
+
+    try {
+      setIsLoading(true);
+      await updateTodoTitle(todo.id, newTitle);
+      title = newTitle;
```

Comment:
It's bad approach. Please find another way to update title in ToDo (if you won't have any ideas, ask in fe_chat)